test/codec_test.dart

```diff
@@ -751,4 +752,21 @@ void main() {
     expect(readUint64([180, 222, 252, 255, 255, 255, 255, 255, 255, 1]),
         expect64(0xffff2f34, 0xffffffff));
   });
+
+  test('testWriteTo', () {
+    var writer = new CodedBufferWriter()..writeField(0, PbFieldType.O3, 1337);
+    expect(writer.lengthInBytes, 3);
+    var rng = new Random();
```

Comment:
I don't like using Random values in a unit test, since it's a gateway to flaky tests (unless you seed the rng with a fixed value).

In this case, you just need sentinel values to make sure the buffer writes don't over/underflow, so a fixed value that's different from the values in `expected` should do the trick.